internal/server/cache/cache.go

```diff
@@ -0,0 +1,67 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+	rate "github.com/go-redis/redis_rate/v9"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+type Cache struct {
+	client *redis.Client
+}
+
+type Options struct {
+	Host string
+	Port int
+	Username string
+	Password string
+	Options string
+}
+
+func NewCache(options Options) *Cache {
+	var client *redis.Client
+
+	if len(options.Host) > 0 {
+		redisOptions, err := redis.ParseURL(fmt.Sprintf("redis://%s:%d?%s", options.Host, options.Port, options.Options))
```

Comment:
Could we accept the url as a single string, instead of separate fields for each? Fewer configuration options makes the server config a little bit easier to document.

Reply:
I generally dislike this kind of configuration as a single string value. It requires the operator to know the connection string format. This format is usually harder to communicate without referencing upstream documentation, in this case https://pkg.go.dev/github.com/go-redis/redis/v8#ParseURL. IMO, for a new-ish operator, it's easier to understand individual `Host`, `Port`, `Username`, `Password` fields than the full connection string.

Additionally infrastructure configurations don't usually come in a neat package: hostname comes from one source, port from another, username/password are user-defined, options are optional and user-specific. This means the user needs to gather all necessary information and compose a connection string from the results. It also means individual fields cannot be updated without changing the entire string which can lead to mistakes and misconfiguration.

I can add the single string value as an alternative but it's not something I would ever use.

Reply:
Ok, that makes sense. I generally prefer a single string, but I also have very little experience with terraform, so I haven't run into those problems. I don't think there's any reason to have two ways to do this right now, so let's keep it as separate fields.